x-pack/plugins/ml/public/alerting/register_ml_alerts.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { lazy } from 'react';
+import { MlStartDependencies } from '../plugin';
+import { ML_ALERT_TYPES } from '../../common/constants/alerts';
+import { MlAnomalyDetectionAlertParams } from '../../common/types/alerts';
+
+export function registerMlAlerts(
+  alertTypeRegistry: MlStartDependencies['triggersActionsUi']['alertTypeRegistry']
+) {
+  alertTypeRegistry.register({
+    id: ML_ALERT_TYPES.ANOMALY_DETECTION,
+    description: i18n.translate('xpack.ml.alertTypes.anomalyDetection.description', {
+      defaultMessage: 'Alert when anomaly detection jobs results match the condition.',
+    }),
+    iconClass: 'bell',
+    documentationUrl(docLinks) {
+      return `${docLinks.ELASTIC_WEBSITE_URL}guide/en/machine-learning/${docLinks.DOC_LINK_VERSION}/ml-configuring-alerts.html`;
+    },
+    alertParamsExpression: lazy(() => import('./ml_anomaly_alert_trigger')),
+    validate: (alertParams: MlAnomalyDetectionAlertParams) => {
+      const validationResult = {
+        errors: {
+          jobSelection: new Array<string>(),
+          severity: new Array<string>(),
+          resultType: new Array<string>(),
+        },
+      };
+
+      if (
+        !alertParams.jobSelection?.jobIds?.length &&
+        !alertParams.jobSelection?.groupIds?.length
+      ) {
+        validationResult.errors.jobSelection.push(
+          i18n.translate('xpack.ml.alertTypes.anomalyDetection.jobSelection.errorMessage', {
+            defaultMessage: 'Job selection is required',
+          })
+        );
+      }
+
+      if (alertParams.severity === undefined) {
+        validationResult.errors.severity.push(
+          i18n.translate('xpack.ml.alertTypes.anomalyDetection.severity.errorMessage', {
+            defaultMessage: 'Anomaly severity is required',
+          })
+        );
+      }
+
+      if (alertParams.resultType === undefined) {
+        validationResult.errors.resultType.push(
+          i18n.translate('xpack.ml.alertTypes.anomalyDetection.resultType.errorMessage', {
+            defaultMessage: 'Result type is required',
+          })
+        );
+      }
+
+      return validationResult;
+    },
+    requiresAppContext: false,
+    defaultActionMessage: i18n.translate(
+      'xpack.ml.alertTypes.anomalyDetection.defaultActionMessage',
+      {
+        defaultMessage: `Elastic Stack Machine Learning Alert:
+- Job IDs: \\{\\{#context.jobIds\\}\\}\\{\\{context.jobIds\\}\\} - \\{\\{/context.jobIds\\}\\}
+- Time: \\{\\{context.timestampIso8601\\}\\}
+- Anomaly score: \\{\\{context.score\\}\\}
+
```

Comment:
Please add an extra sentence in here. @szabosteve may want to edit this text.

```
Alerts are raised based on real-time scores. Remember that scores may be adjusted over time as data continues to be analyzed.
```

![image](https://user-images.githubusercontent.com/7405507/107360365-bad08600-6acd-11eb-814e-79ce068d0ce9.png)

Reply:
@peteharverson shall I add this sentence only when interim results are presented?

Reply:
No this sentence should always be shown. Scores may also change subsequently due to renormalization.

Reply:
Added here a19a6ce84d3e320b8bbcbe827283bcb14404db07

Reply:
The check on `isInterim` is still there in https://github.com/elastic/kibana/commit/a19a6ce84d3e320b8bbcbe827283bcb14404db07

Reply:
Check removed in 513298e27dd6822b1737f4050046c65be1e32ffb